### PR TITLE
MAISTRA-1160 Add documentation on how to use an existing Jaeger instance

### PR DESCRIPTION
**Pull request by dgn**

### CODE REVIEW

**Comment by dgn** [comment]
@neal-timpe maybe you can take a look; we should also have a similar section in the product docs

**Comment by FilipB** [comment]
@dgn Would it be possible to add a description of how to get the `htpasswd` secret (a secret mentioned in docs: Maistra uses a secret called `htpasswd`)? This secret should be added to jaeger instance but the doc does not say how to get it from maistra.

**Comment by dgn** [comment]
> @dgn Would it be possible to add a description of how to get the `htpasswd` secret (a secret mentioned in docs: Maistra uses a secret called `htpasswd`)? This secret should be added to jaeger instance but the doc does not say how to get it from maistra.

Maybe 'uses' is the wrong term. The operator will create that secret in the control plane namespace.

**Comment by FilipB** [comment]
@dgn I got it now, thank you. So since the jaeger is running in the same namespace as the Maistra control plane, it will find the secret called `htpasswd` created by maistra operator. No need for manual creation. It would be nice to also describe this in the doc.

**Comment by dgn** [comment]
> @dgn I got it now, thank you. So since the jaeger is running in the same namespace as the Maistra control plane, it will find the secret called `htpasswd` created by maistra operator. No need for manual creation.

Exactly

**Comment by dgn** [comment]
I added instructions on how to retrieve and set the Jaeger dashboard URL

**Comment by neal-timpe** [comment]
@JStickler said she is looking into it.

**Review by knrc — APPROVED** [review]

**Comment by brian-avery** [comment]
General comment here. I will highlight what stands out to me, but this should be following our documentation standard: https://github.com/openshift/openshift-docs/blob/master/contributing_to_docs/doc_guidelines.adoc#writing-concepts

File: topics/docs/monitoring_and_tracing/using-existing-jaeger.adoc [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+description: "See how to use an existing Jaeger instance for trace collection and querying"
+---
+
+:imagesdir: ../../images
+
+== Configuring Jaeger
+
+=== Deploy to the Control Plane Namespace
+
+In order for a custom Jaeger instance to work with a Maistra control plane, it has to be deployed to the same namespace. When using the Jaeger Operator, this is achieved by creating the Jaeger resource in the same namespace as the ServiceMeshControlPlane resource.
```
Comment (brian-avery) [inline topics/docs/monitoring_and_tracing/using-existing-jaeger.adoc]:
```suggestion
In order for a custom Jaeger instance to work with a Maistra control plane, it has to be deployed to the same namespace. When using the Jaeger Operator, this is achieved by creating the Jaeger resource in the same namespace as the `ServiceMeshControlPlane` resource.
```

File: topics/docs/monitoring_and_tracing/using-existing-jaeger.adoc [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+=== Set Endpoint for Trace Collection
+
+Maistra gateways and sidecars send traces to Mixer (the `istio-telemetry` pod in the control plane namespace), which then passes them on to a `jaeger-collector` endpoint in your cluster. When using a custom Jaeger instance, you will have to make sure to supply the correct endpoint for Mixer to use. This is achieved by setting `spec.istio.global.tracer.zipkin.address` to the hostname and port of your `jaeger-collector` service. The hostname of that service is usually `<jaeger-instance-name>-collector.<namespace>.svc`.
+
+=== Set Endpoint for Trace Querying
+
+If you're using Kiali to visualize traces and metrics across your mesh, you will have to make sure that Kiali knows from which Jaeger instance it should gather traces. You do this by setting `spec.istio.kiali.jaegerInClusterURL` to the hostname of your `jaeger-query` service - the port is normally not required, as it uses 443 by default. As above, the hostname of that service is usually `<jaeger-instance-name>-query.<namespace>.svc`.
```
Comment (brian-avery) [inline topics/docs/monitoring_and_tracing/using-existing-jaeger.adoc:63]: jaegerInClusterURL should be in the installation options

File: topics/docs/monitoring_and_tracing/using-existing-jaeger.adoc [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+$ oc get route -n istio-system external-jaeger
+NAME                   HOST/PORT                                     PATH   SERVICES               [...]
+external-jaeger        external-jaeger-istio-system.apps-crc.testing        external-jaeger-query  [...]
+```
+
+The value under `HOST/PORT` is the externally accessible URL of the Jaeger dashboard.
+
+=== Example SMCP Resource
+
+The following ServiceMeshControlPlane object assumes that you have deployed Jaeger using the Jaeger Operator and the above Jaeger resource.
```
Comment (brian-avery) [inline topics/docs/monitoring_and_tracing/using-existing-jaeger.adoc]:
```suggestion
The following `ServiceMeshControlPlane` object assumes that you have deployed Jaeger using the Jaeger Operator and the above Jaeger resource.
```

File: topics/docs/monitoring_and_tracing/using-existing-jaeger.adoc (hunk 2 above)
Comment (dgn) [reply topics/docs/monitoring_and_tracing/using-existing-jaeger.adoc:63]: good point, done!

**Review by brian-avery — APPROVED** [review]

**Pull request merged**